foodji_api/Application/Queries/GetRecipeByIdQuery.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Application.Dto;
+using AutoMapper;
+using Domain.Recipes;
+using Infra;
+using MediatR;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace Application.Queries;
+
+public class GetRecipeByIdQuery : IRequest<RecipeDto?>
+{
+    public ObjectId RecipeId { get; }
+
+    public GetRecipeByIdQuery(string recipeId)
+    {
+        RecipeId = new ObjectId(recipeId);
+    }
+    
+    private class Handler : IRequestHandler<GetRecipeByIdQuery, RecipeDto?>
+    {
+        private readonly IFoodjiDbClient _client;
+        private readonly IMapper _mapper;
+
+        public Handler(IFoodjiDbClient client, IMapper mapper)
+        {
+            _client = client;
+            _mapper = mapper;
+        }
+
+
+        public async Task<RecipeDto?> Handle(GetRecipeByIdQuery request, CancellationToken cancellationToken)
+        {
+            var results = await _client.Recipes.FindAsync(x => x.Id == request.RecipeId, cancellationToken: cancellationToken);
+
+            var recipes = results.ToList();
+            
+            if (recipes.Count > 1)
+            {
+                // TODO More specific exception to go along better exception handling in the API layer
+                //      500 many with the same ID (bad news!)
+                //      shouldn't happen, but no "FindOne" method to make that check for us
+                throw new Exception($"{recipes.Count} recipes with the id {request.RecipeId}");
+            }
+
+            if (recipes.Count == 0)
+            {
+                return null;
```

Comment:
null ou on throw une exception?

Reply:
Null, c'est l'API qui gère en envoyant un 404 pour not found, et c'est généralement moins "expensive" de faire un null check qu'un throw/catch.

Reply:
(À noté que si il y a plus qu'une recette avec cet ID, je traite effectivement comme une erreur, parce que là, on a un sérieux problème de BD à l'interne, qui n'a rien à voir avec si l'ID lui-même est valide - as in existe pour une recette)